src/templates/speciality-component.js

```diff
@@ -0,0 +1,118 @@
+import React, { Fragment } from 'react'
+import Ajv from 'ajv'
+
+import {
+  communitySchema,
+  projectsSchema,
+  clientSchema
+} from './speciality/schemas'
+
+import {
+  getExternalType,
+  getSpecialityEvents,
+  flattenSpeciality
+} from './speciality/methods'
+
+import IntroSection from '../components/Speciality/Intro'
+import ProjectsSection from '../components/Speciality/Projects'
+import TrainingSection from '../components/Speciality/Training'
+import CommunitySection from '../components/Speciality/Community'
+import EventSection from '../components/Common/Events'
+import TalksSection from '../components/Speciality/Talks'
+import GetInTouch from '../components/Common/GetInTouch'
+import TutorialsSection from '../components/Speciality/Tutorials'
+import BooksSection from '../components/Speciality/Books'
+import BlogListing from '../components/Common/BlogListing'
+import GreyBackground from '../components/Common/GreyBackground'
+
+const ajv = new Ajv({ allErrors: true })
+
+export const SpecialityView = props => {
+  const {
+    data: {
+      contentfulSpeciality: speciality,
+      allContentfulMeetupEvent: { edges: events },
+      videoIcon,
+      filteredPosts = []
+    }
+  } = props
+
+  const flattenedSpeciality = flattenSpeciality(speciality)
+
+  const validateCommunity = ajv.compile(communitySchema)
+  const validateProjects = ajv.compile(projectsSchema)
+  const validateClients = ajv.compile(clientSchema)
+
+  const {
+    title,
+    contactText,
+    eventIcon,
+    relatedProjects,
+    clients
+  } = flattenedSpeciality
+
+  const { edges: postEdges = [] } = filteredPosts
+  const posts = postEdges.map(({ node }) => node)
+  const talks = getExternalType(flattenedSpeciality, `Talk`)
+  const tutorials = getExternalType(flattenedSpeciality, `Tutorial`)
+  const books = getExternalType(flattenedSpeciality, `Book`)
+  const specialityEvents =
+    events && events.length > 0 ? getSpecialityEvents(title, events) : []
+
+  const renderProjectsSection =
+    validateProjects(relatedProjects) || validateClients(clients)
+  const renderCommunitySection = validateCommunity(flattenedSpeciality)
+
+  const renderTalksSection = talks && talks.length > 0
```

Comment:
I would have called all these sections `hasTalks`, `hasBooks` and `hasBlogs` even if it's pretty easy to understand what it does.